src/celeritas/neutron/executor/NeutronElasticExecutor.hh

```diff
@@ -0,0 +1,71 @@
+//----------------------------------*-C++-*----------------------------------//
+// Copyright 2024 UT-Battelle, LLC, and other Celeritas developers.
+// See the top-level COPYRIGHT file for details.
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file celeritas/neutron/executor/NeutronElasticExecutor.hh
+//---------------------------------------------------------------------------//
+#pragma once
+
+#include "corecel/Assert.hh"
+#include "celeritas/global/CoreTrackView.hh"
+#include "celeritas/mat/ElementSelector.hh"
+#include "celeritas/mat/ElementView.hh"
+#include "celeritas/mat/IsotopeSelector.hh"
+#include "celeritas/mat/IsotopeView.hh"
+#include "celeritas/neutron/data/NeutronElasticData.hh"
+#include "celeritas/neutron/interactor/NeutronElasticInteractor.hh"
+#include "celeritas/neutron/xs/NeutronElasticMicroXsCalculator.hh"
+#include "celeritas/phys/Interaction.hh"
+
+namespace celeritas
+{
+//---------------------------------------------------------------------------//
+struct NeutronElasticExecutor
+{
+    inline CELER_FUNCTION Interaction
+    operator()(celeritas::CoreTrackView const& track);
+
+    NeutronElasticRef params;
+};
+
+//---------------------------------------------------------------------------//
+/*!
+ * Apply the NeutronElasticInteractor to the current track.
+ */
+CELER_FUNCTION Interaction
+NeutronElasticExecutor::operator()(CoreTrackView const& track)
+{
+    auto particle = track.make_particle_view();
+    auto const& dir = track.make_geo_view().dir();
+    auto rng = track.make_rng_engine();
+
+    // Select a target element
+    auto material = track.make_material_view().make_material_view();
```

Comment:
This funny line makes me think that should rename the first function call to `make_material_track_view()`.

Reply:
Yeah I've also had a plan for refactoring some of the code underpinning views so that we can have just a single "material view". This was complicated by the fact that we needed a CPU-only material view that didn't have the full material state...